XamlAnimatedGif/AnimationBehavior.cs

```diff
@@ -158,6 +158,27 @@ public static void SetAnimateInDesignMode(DependencyObject obj, bool value)
 
         #endregion
 
+        #region DownloadCacheLocation
+
+        public static string GetDownloadCacheLocation(DependencyObject obj)
+        {
+            return (string)obj.GetValue(DownloadCacheLocationProperty);
+        }
+
+        public static void SetDownloadCacheLocation(DependencyObject obj, string value)
+        {
+            obj.SetValue(DownloadCacheLocationProperty, value);
+        }
+
+        public static readonly DependencyProperty DownloadCacheLocationProperty =
+            DependencyProperty.RegisterAttached(
+                "DownloadCacheLocation",
+                typeof(string),
+                typeof(AnimationBehavior),
+                new PropertyMetadata(Path.GetTempPath()));
+
+        #endregion
```

Comment:
As mentioned in the issue, there's no need for an attached property here. It doesn't need to be configured for each image control (in fact, it would be pretty annoying to have to specify it for each image control).

Just a static field would be fine for this (probably on the `UriLoader` class, since it's where it will be used).

This should make most of the other changes unnecessary.